clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ResetPositionsApplicationEvent.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.clients.consumer.internals.events;
+
+/**
+ * Event for resetting offsets for all assigned partitions that require it. This is an
+ * asynchronous event that generates ListOffsets requests, and completes by updating in-memory
+ * positions when responses are received.
+ */
+public class ResetPositionsApplicationEvent extends CompletableApplicationEvent<Void> {
+
+    public ResetPositionsApplicationEvent() {
+        super(Type.RESET_POSITIONS);
+    }
+}
```

Comment:
Can you add `toString()`, `hashCode()`, and `equals()` methods here? Events appear in logs and there are places in the code (maybe just tests?) that check event equality.

Reply:
It's probably useful for this event to print with `ResetPositionsApplicationEvent` instead of `CompletableApplicationEvent`. We could still customize the `toString` method here by reusing a util method in the parent class.

Reply:
Sounds good. I updated the toString in `CompletableApplicationEvent` to include the getClass().getSimpleName(), so we end up getting `ResetPositionsApplicationEvent` (or any other child class) without having to redefine the toString()

Reply:
@kirktrue regarding hashCode and equals, this `ResetPositionsApplicationEvent` class (and the ValidatePositions one) do not have any extra attributes, so the base hashCode and equals defined in the parent `CompletableApplicationEvent` should be enough right? Am I missing something?